pkg/cli/image/extract/extract.go

```diff
@@ -375,17 +386,33 @@ func (o *ExtractOptions) Run() error {
 					return fmt.Errorf("unable to connect to image repository %s: %v", from.String(), err)
 				}
 
-				srcManifest, location, err := imagemanifest.FirstManifest(ctx, from.Ref, repo, o.FilterOptions.Include)
+				srcManifest, location, err := retrieveSourceManifest(ctx, from, repo, o)
 				if err != nil {
-					if imagemanifest.IsImageForbidden(err) {
-						msg := fmt.Sprintf("image %q does not exist or you don't have permission to access the repository", from)
-						return imagemanifest.NewImageForbidden(msg, err)
-					}
-					if imagemanifest.IsImageNotFound(err) {
-						msg := fmt.Sprintf("image %q not found: %s", from, err.Error())
-						return imagemanifest.NewImageNotFound(msg, err)
+					if strings.Contains(err.Error(), "a manifest image could not be found for this platform") {
+						// Translate the current runtime OS to linux when looking through a manifest since there will not be a darwin/'archType' in a manifest image.
+						// In cases where the runtime OS is windows or darwin (i.e. macOS), there is no expectation of a manifest image with those particular OS's.
+						if o.FilterOptions.DefaultOSFilter {
+							runtimeOS := runtime.GOOS
+							arch := runtime.GOARCH
+							if linuxConversionWarning == "" {
```

Comment:
There is no need to this. Warn user and continue?

Reply:
This is the implementation solution for
```
If user passes --filter-by-os flag explicitly, we will use it and if it is not found we return error without any defaulting because user deliberately wants that image but we don't have it(for example --filter-by-os=darwin/amd64)
If user does not pass anything
```
in https://github.com/openshift/oc/pull/1311#issuecomment-1471799668. This checks if `--filter-by-os` was passed in or not.